From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: chickeneer <emcchickeneer@gmail.com>
Date: Sat, 10 Jul 2021 23:28:53 -0500
Subject: [PATCH] Add VegetationPatchPlaceEvent for bonemealing Moss Block

Moss Blocks that are bonemealed can change nearby blocks
not just add new blocks - this is a patch to cancel that action

diff --git a/src/main/java/com/empireminecraft/customevents/VegetationPatchPlaceEvent.java b/src/main/java/com/empireminecraft/customevents/VegetationPatchPlaceEvent.java
new file mode 100644
index 0000000000000000000000000000000000000000..f011836072670141ae5291e0b58346ad2da98b8f
--- /dev/null
+++ b/src/main/java/com/empireminecraft/customevents/VegetationPatchPlaceEvent.java
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2021. Starlis LLC / dba Empire Minecraft
+ *
+ * This source code is proprietary software and must not be redistributed without Starlis LLC's approval
+ *
+ */
+
+package com.empireminecraft.customevents;
+
+import org.bukkit.block.Block;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.block.BlockEvent;
+import org.jetbrains.annotations.NotNull;
+
+public class VegetationPatchPlaceEvent extends BlockEvent implements Cancellable {
+    private static final HandlerList handlers = new HandlerList();
+    private final Block origin;
+    private boolean cancelled;
+
+    public VegetationPatchPlaceEvent(@NotNull Block block, @NotNull Block origin) {
+        super(block);
+        this.origin = origin;
+    }
+
+    @NotNull
+    public Block getOrigin() {
+        return this.origin;
+    }
+
+    @Override
+    public boolean isCancelled() {
+        return cancelled;
+    }
+
+    @Override
+    public void setCancelled(boolean cancelled) {
+        this.cancelled = cancelled;
+    }
+
+    @NotNull
+    @Override
+    public HandlerList getHandlers() {
+        return handlers;
+    }
+
+    @NotNull
+    public static HandlerList getHandlerList() {
+        return handlers;
+    }
+}
